lib/java_buildpack/framework/app_dynamics_agent.rb

```diff
@@ -60,9 +60,10 @@ def supports?
       private_constant :FILTER
 
       def application_name(java_opts, credentials)
-        name = credentials['application-name'] || @configuration['default_application_name'] ||
-          @application.details['application_name']
+        name = ENV['APPDYNAMICS_APPLICATION_NAME'] || credentials['application-name']
```

Comment:
This is an unneeded change as the introduction of `APPDYNAMICS_APPLICATION_NAME` is redundant.  Configuration of the the application name can already be done in three other ways.